File: LC-1389/LC-1389.js
/**
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
var createTargetArray = function (nums, index) {
  /**
   * Concepts: Naive Approach
   * Use Array.prototype.splice function
   */

  let result = [],
    numsLen = nums.length
  for (let i = 0; i < numsLen; ++i) {
    result.splice(index[i], 0, nums[i])
  }
  return result
}

File: LC-1389/LC-1389.cpp
class Solution {
public:
    /**
     * Concepts: Naive Approach
     * Use vector::insert function
     */
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> result;
        vector<int>::iterator it;
        int numsLen = nums.size();
        for (int i = 0; i < numsLen; ++i) {
            it = result.begin() + index[i];
            result.insert(it, nums[i]);
        }
        return result;
    }
};